### PR TITLE
add function:filterByOverlappingRegions

### PR DESCRIPTION
**Pull request by liamlee**
for multiple regions data select.

### CODE REVIEW

**Comment by heuermh** [comment]
Jenkins, test this please

**Comment by heuermh** [comment]
From the failed Jenkins build
```
[INFO] Compiling 71 source files to /home/jenkins/workspace/ADAM-prb/HADOOP_VERSION/2.6.0/SCALAVER/2.10/SPARK_VERSION/1.5.2/label/centos/adam-core/target/2.10.6/test-classes at 1479736143868
[ERROR] /home/jenkins/workspace/ADAM-prb/HADOOP_VERSION/2.6.0/SCALAVER/2.10/SPARK_VERSION/1.5.2/label/centos/adam-core/src/test/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDDSuite.scala:110: error: not found: value resourcePath
[ERROR]     val inputPath = resourcePath("artificial.sam")
[ERROR]                     ^
[ERROR] one error found
```

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala [hunk 1]
```diff
@@ -366,6 +366,24 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] {
...
+   *   querys region.
+   */
+  def filterByOverlappingRegions(querys: List[ReferenceRegion]): U = {
+    replaceRdd(rdd.filter(elem => {
+
+	  val regions = getReferenceRegions(elem)
+
+	  querys.map(query => {
+	    regions.exists(_.overlaps(query))
+	  }).reduce((a, b) => a || b)
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala]: I would replace `reduce((a, b) => a || b)` with `fold(false)((a, b) => a || b)` as `reduce` will fail on an empty collection. You can get an empty result from `getReferenceRegions(elem)` with unmapped reads.

File: adam-core/src/test/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDDSuite.scala [hunk 2]
```diff
@@ -106,6 +106,16 @@ class AlignmentRecordRDDSuite extends ADAMFunSuite {
     assert(coverage.rdd.filter(r => r.start == 30).first.count == pointCoverage)
   }
 
+  sparkTest("test filterByOverlappingRegions") {
+    val inputPath = resourcePath("artificial.sam")
```
Comment (fnothaft) [inline adam-core/src/test/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDDSuite.scala]: You'll want to change `resourcePath("artificial.sam")` to `testFile("artificial.sam")` and then this should compile.

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala (hunk 1 above)
Comment (liamlee) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala]: This is a very helpful advise, Thanks!

**Comment by fnothaft** [comment]
Jenkins, test this please.

**Comment by fnothaft** [comment]
Merged! Thank you very much @liamlee for the contribution!

**Comment by liamlee** [comment]
My pleasure.

**Pull request merged**